### ISSUE TITLE
Chocolatey Support

### ISSUE THREAD

**Issue by D-Maxwell** [huynhsontung/Screenbox#399]
**Is your feature request related to a problem? Please describe.**
Some may be unable to install from the msstore source or may prefer not to deal with its cumbersomeness.

**Describe the solution you'd like**
Having read the constraints and other restrictions that winget had in place regarding package submission, I wholeheartedly understand your decision not to spend such an amount in something that, in my honest opinion, you should have never been required to pay for in the first place.
That is why I am instead proposing the idea of submitting this to an alternative repository, such as that which Chocolatey offers.

**Describe alternatives you've considered**
Having an auto-update function built in could easily imply a lot of work whilst also being redundant, as some may prefer to update the package through external tools, be those a wrapper for the msstore repository, or, should it be conceivable, a winget utility such as WingetUI.

**Additional context**
As things currently stand, packages for Screenbox are only available (through winget) from the msstore repository. As the latter is notoriously painful to make work, at least as I for one have got to experience --for a myraid of reasons, including but certainly (oh god no) not limited to windows builds not being up to date, as per Microsoft's almighty arbitrary stances on the meaning of "up to date", but enough with the rambling-- each of my machines, to this day, lacks support for this specific source, due to its heavy requirements.

**Comment by United600** [comment]
If Chocolatey supports msix, anyone (I think, not sure) can add the sideload version from the release tab, obviously it would be better if it was the signed store version.
Hopefully someone with more knowledge of how it works can submit and maintain it.

> limited to windows builds not being up to date, as per Microsoft's almighty arbitrary stances on the meaning of "up to date", but enough with the rambling-- each of my machines, to this day, lacks support for this specific source, due to its heavy requirements.

Is there really any supported version of consumer Windows that doesn't support Winget out of the box?

**Comment by huynhsontung** [comment]
I briefly looked at Chocolatey and if I understand correctly, like winget, it can download and run the installers silently to install applications. So technically, I can create a Chocolatey package that downloads the sideload package from GitHub and runs the silent install script as Administrator. Users with Chocolatey will have auto-updates for sideload packages as well. That's pretty neat.

Like @United600 mentioned, it would be better if the package is properly signed. Without it, every update will install a new certificate to the user's trusted CA store. Unfortunately, a trusted CA certificate will cost me an arm and a leg. I'm not sure what's Chocolatey's stand on unsigned packages or installing self-signed certs onto the user machine. I can't find those details anywhere.

Other than that, I am all for supporting Chocolatey.

**Comment by huynhsontung** [comment]
Or actually, I can download the signed package directly from MS Store and redistribute it as well. If that works reliably, I will try to submit the store signed packages to both winget and chocolatey.

**Comment by D-Maxwell** [comment]
> If Chocolatey supports msix, anyone (I think, not sure) can add the sideload version from the release tab, obviously it would be better if it was the signed store version. Hopefully someone with more knowledge of how it works can submit and maintain it.
> 
> > limited to windows builds not being up to date, as per Microsoft's almighty arbitrary stances on the meaning of "up to date", but enough with the rambling-- each of my machines, to this day, lacks support for this specific source, due to its heavy requirements.
> 
> Is there really any supported version of consumer Windows that doesn't support Winget out of the box?

Yes. I cannot even begin to recall the amount of devices I've had where winget was either not updated and had compatibility issues with its own sources, or outright refused to function because it detected a "not up to date" build of Windows, or, most commonly, would not satisfy the requirements imposed by Microsoft's servers after a debloat and would therefore be refused any connections.

**Comment by huynhsontung** [comment]
Totally forgot to update on this issue but Screenbox is officially on winget!

`winget install Screenbox -s winget`

I'm not sure about Chocolatey. I remember looking into Chocolatey documentation and I were testing with it for a while and concluded it's not worth it.  Not all dependencies are available on Chocolatey so I can't submit my package there. I may be misremembering.
